### PR TITLE
feat: prune origin during switch

### PR DESCRIPTION
**Pull request by mcwarman**
## Motivation
Remove the gone remotes during switch back to default after the merge.

## Changes

- add remote prune before delete
- switch using switch over checkout

### CODE REVIEW

File: bb-pr [hunk 1]
```diff
@@ -319,8 +319,9 @@ action_checkout() {
 
 git_switch_default() {
   if [[ -n "$GIT_REMOTE_DEFAULT" && "$GIT_LOCAL_WORKING_BRANCH" ]]; then
-    git checkout "$GIT_REMOTE_DEFAULT"
+    git switch "$GIT_REMOTE_DEFAULT"
     git pull
+    git remote prune origin
```
Comment (mcwarman) [inline bb-pr:324]: Maybe prune isn't the correct answer:

```
URL: git@bitbucket.org:....git
 * [pruned] origin/feat/...
 * [pruned] origin/feat/....
 * [pruned] origin/feat/...
 * [pruned] origin/feat/...
 * [pruned] origin/feature/...
Deleted branch feat/... (was xxxx).
````

Comment (quotidian-ennui) [reply bb-pr:324]: Might be best to leave it as an exercise for the user to do a `git fetch origin --prune --prune-tags`

It'll be that 1% edge case where you really burn yourself.

**Review by quotidian-ennui — APPROVED** [review]

File: bb-pr (hunk 1 above)
Comment (mcwarman) [reply bb-pr:324]: @quotidian-ennui What about:

```shell
git branch --remotes --delete $(git rev-parse --abbrev-ref --symbolic-full-name @{u})
```

**Comment by mcwarman** [comment]
@quotidian-ennui fix worked as expected:

```
Deleted remote-tracking branch origin/refactor/.... (was xxx).
Deleted branch refactor/.... (was xxx).
```

**Review by quotidian-ennui — APPROVED** [review]

**Pull request merged**